toolchain/check/handle_class.cpp

```diff
@@ -346,15 +345,18 @@ static auto DiagnoseClassSpecificDeclRepeated(Context& context,
                                               SemIRLoc prev_loc,
                                               Lex::TokenKind tok) -> void {
   CARBON_DIAGNOSTIC(ClassSpecificDeclRepeated, Error,
-                    "Multiple `{0}` declarations in class.{1}", Lex::TokenKind,
-                    std::string);
-  const llvm::StringRef extra = tok == Lex::TokenKind::Base
-                                    ? " Multiple inheritance is not permitted."
-                                    : "";
+                    "multiple `{0}` declarations in class", Lex::TokenKind);
+  CARBON_DIAGNOSTIC(BaseDeclRepeated, Error,
+                    "multiple `{0}` declarations in class; multiple "
+                    "inheritance is not permitted",
+                    Lex::TokenKind);
   CARBON_DIAGNOSTIC(ClassSpecificDeclPrevious, Note,
-                    "Previous `{0}` declaration is here.", Lex::TokenKind);
+                    "previous `{0}` declaration is here", Lex::TokenKind);
   context.emitter()
-      .Build(new_loc, ClassSpecificDeclRepeated, tok, extra.str())
+      .Build(new_loc,
+             tok == Lex::TokenKind::Base ? BaseDeclRepeated
+                                         : ClassSpecificDeclRepeated,
+             tok)
       .Note(prev_loc, ClassSpecificDeclPrevious, tok)
       .Emit();
```

Comment:
```suggestion
  CARBON_DIAGNOSTIC(ClassSpecificDeclPrevious, Note,
                    "previous `{0}` declaration is here", Lex::TokenKind);
  if (tok == Lex::TokenKind::Base) {
    CARBON_DIAGNOSTIC(BaseDeclRepeated, Error,
                      "multiple `base` declarations in class; multiple "
                      "inheritance is not permitted");
    context.emitter()
        .Build(new_loc, BaseDeclRepeated)
        .Note(prev_loc, ClassSpecificDeclPrevious, tok)
        .Emit();
  } else {
    CARBON_DIAGNOSTIC(ClassSpecificDeclRepeated, Error,
                      "multiple `{0}` declarations in class", Lex::TokenKind);
    context.emitter()
        .Build(new_loc, ClassSpecificDeclRepeated, tok)
        .Note(prev_loc, ClassSpecificDeclPrevious, tok)
        .Emit();
  }
```

Suggesting to drop the token argument when the type is known. This felt like the shortest approach on that route... (note, delta isn't quite correct, but I don't think I can get git to do better)